web/modules/custom/joinup_federation/src/Plugin/pipeline/Step/UserSelectionFilter.php

```diff
@@ -381,6 +425,7 @@ protected function getCategory(RdfInterface $activity): string {
    */
   protected function getInfo(string $id, string $category): MarkupInterface {
     $activity = $this->provenanceHelper->loadOrCreateEntityActivity($id);
+    $collection = $this->provenanceHelper->loadActivityAssociatedEntity($activity);
```

Comment:
No, I also think that we should not make it an internal URI. Looking at our code, I don't think we are going to have a case where it is not a local entity, however, it might not be that in the future.
I am going to change the `::loadActivityAssociatedEntity` to `::loadActivityAssociateWith` to return a URI or the entity URL and will remove the label of the entity.